pkg/bootstrapper/node_bootstrapper.go

```diff
@@ -10,38 +34,382 @@ package bootstrapper
+		}
+	}
+	// Populate destination directory with the files we need
+	if wcb.ignitionFilePath != "" {
+		err = wcb.parseIgnitionFile(wcb.ignitionFilePath, filesToTranslate)
+		if err != nil {
+			return fmt.Errorf("could not parse ignition file: %s", err)
+		}
+	}
+	return nil
 }
 
-// Start kubelet as a Windows service to ensure that kubelet is automatically started when it fails
-func (wcb *winNodeBootstrapper) startKubeletService() bool {
-	return false
+// createKubeletService creates a new kubelet service to our specifications
+func (wcb *winNodeBootstrapper) createKubeletService() error {
+	var err error
+	kubeletArgs := []string{
+		"--config=" + wcb.kubeletConfPath,
+		"--bootstrap-kubeconfig=" + filepath.Join(wcb.installDir, "bootstrap-kubeconfig"),
+		"--kubeconfig=" + wcb.kubeconfigPath,
+		"--pod-infra-container-image=mcr.microsoft.com/k8s/core/pause:1.2.0",
+		"--cert-dir=c:/var/lib/kubelet/pki/",
+		"--windows-service",
+		"--logtostderr=false",
+		"--log-file=" + filepath.Join(wcb.installDir, "kubelet.log"),
+		// TODO: Uncomment this when we have a CNI solution
+		/*
+			network-plugin=cni",
+			cni-bin-dir=" + filepath.Join(k8sInstallDir, "cni"),
+			cni-conf-dir=" + filepath.Join(k8sInstallDir, "cni"),
+		*/
+	}
+	if cloudProvider, ok := wcb.kubeletArgs["cloud-provider"]; ok {
+		kubeletArgs = append(kubeletArgs, "--cloud-provider="+cloudProvider)
+	}
+	if v, ok := wcb.kubeletArgs["v"]; ok {
+		kubeletArgs = append(kubeletArgs, "--v="+v)
+	}
+	// Mostly default values here
+	c := mgr.Config{
+		ServiceType: 0,
+		// StartAutomatic will start the service again if the node restarts
+		StartType:    mgr.StartAutomatic,
+		ErrorControl: 0,
+		// Path to kubelet.exe
+		BinaryPathName:   filepath.Join(wcb.installDir, "kubelet.exe"),
+		LoadOrderGroup:   "",
+		TagId:            0,
+		Dependencies:     nil,
+		ServiceStartName: "",
+		DisplayName:      "",
+		Password:         "",
+		Description:      "",
+	}
+	wcb.kubeletSVC, err = wcb.svcMgr.CreateService(KubeletServiceName, filepath.Join(wcb.installDir, "kubelet.exe"), c, kubeletArgs...)
+	if err != nil {
+		return err
+	}
+	err = wcb.kubeletSVC.SetRecoveryActions([]mgr.RecoveryAction{
+		{Type: mgr.ServiceRestart, Delay: 5},
+	}, 600)
+	if err != nil {
+		return err
+	}
+	return nil
 }
 
-// TODO:  Add other functions as deemed necessary
+// startKubeletService starts the kubelet as a Windows service
+func (wcb *winNodeBootstrapper) startKubeletService() error {
+	if wcb.kubeletSVC == nil {
+		return fmt.Errorf("no kubelet service")
+	}
+	err := wcb.kubeletSVC.Start()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (wcb *winNodeBootstrapper) removeKubeletService() error {
+	if wcb.kubeletSVC == nil {
+		return nil
+	}
+	return wcb.kubeletSVC.Delete()
+}
+
+// controlService sends a signal to the service and waits until it changes state in response to the signal
+func (wcb *winNodeBootstrapper) controlService(c svc.Cmd, to svc.State) error {
```

Comment:
Please don't use single letter variables. `cmd` is small enough. I also don't follow what this function is doing. Is waiting until it reaches the `to` state? If so you need to mention that in the doc. I also don't understand why this function is called `controlService`.